### ISSUE TITLE
Generate const array of possible enum values

### ISSUE THREAD

**Issue by gwax** [supabase/postgres-meta#832]
**Is your feature request related to a problem? Please describe.**

I would like to have values matching possible enum values that are synchronized to the database schema for looping or use with zod and similar tools.

**Describe the solution you'd like**

for a schema:
```typescript
export type Database = {
  public: {
    Enums: {
      mytype: "value" | "other"
    }
  }
}
```

I would like to have generated values:
```typescript
export const DatabaseValues = {
  public: {
    Enums: {
      mytype: ["value", "other"],
    },
  },
} as const;
```

this would allow defining zod types to match like:

```typescript
const mySchema = z.object({
  mytype: z.enum(DatabaseValues.public.Enums.mytype),
});
```

**Describe alternatives you've considered**

- manually keeping values in sync with the database and hoping they don't fall out of sync
- writing custom type checkers to ensure that values match the schema

**Additional context**

**Comment by toBeOfUse** [comment]
No, you are generating unions of string literals, which is a different thing. The current types that are generated can only be used for static type-checking (unlike proper Typescript enums.)

**Comment by gwax** [comment]
@sweatybridge , could you please reopen this feature request

As @toBeOfUse notes, the code does not generate typescript enums (which are values), it generates string literal unions (which are types). (`export type` vs `export const`)

The types generated currently are removed by the typescript compiler and cannot be used at runtime.

Generated values are useful both for the zod case and for iterating over values.

**Comment by toBeOfUse** [comment]
A workaround is present in #864

**Comment by soedirgo** [comment]
Reopened this gauge interest on generating values in addition to types.

**Comment by toBeOfUse** [comment]
This is a kind of a duplicate of #864 now

**Comment by toBeOfUse** [comment]
#901 looks good to me

**Comment by gwax** [comment]
@soedirgo , #901 does this
